src/sysext/sysext.c

```diff
@@ -2208,6 +2209,46 @@ static int vl_method_refresh(sd_varlink *link, sd_json_variant *parameters, sd_v
         return sd_varlink_reply(link, NULL);
 }
 
+static int parse_version(const char *filename, char **version) {
+    char *start, *last_dot_after_underscore;
+
+    // Find the last occurrences of '_'. This will mark the start of the extension version.
+    start = strrchr(filename, '_');
+    if (!start) {
+        // If no underscore found, assign "none" and return success.
+        // This is for cases where there are no versions in the filename like 'image.raw'.
+        *version = strdup("none");
+        return *version ? 0 : -1;
```

Comment:
Please return specific error code on failure, and update `ret` only when the function succeeds.
```
char *t = strdup("none");
if (!t)
        return -ENOMEM;

*ret = t;
return 0;
```